src/services/armService.ts

```diff
@@ -102,6 +102,12 @@ export class ArmService extends BaseService {
       return;
     }
 
+    for (const key of Object.keys(deployment.parameters)) {
+      if (!deployment.parameters[key].value) {
+        deployment.parameters[key] = undefined;
```

Comment:
We can delete the param using the following: `delete deployment.parameters[key];`